src/config_store.rs

```diff
@@ -82,26 +88,50 @@ impl ConfigStore {
     pub fn add_config(&mut self, config: Arc<CheckConfig>) {
         self.configs.insert(config.subscription_id, config.clone());
 
+        let bucket = self
+            .partitioned_buckets
+            .entry(config.partition)
+            .or_insert_with(|| (0..MAX_CHECK_INTERVAL_SECS).map(|_| vec![]).collect());
+
         // Insert the configuration into the appropriate slots
         for slot in config.slots() {
-            self.buckets[slot].insert(config.subscription_id);
+            bucket[slot].push(config.clone());
         }
     }
 
     /// Remove a Check Configuration from the store.
     pub fn remove_config(&mut self, subscription_id: Uuid) {
-        if let Some(config) = self.configs.remove(&subscription_id) {
-            for slot in config.slots() {
-                self.buckets[slot].remove(&subscription_id);
-            }
+        let Some(config) = self.configs.remove(&subscription_id) else {
+            return;
+        };
+        let Some(buckets) = self.partitioned_buckets.get_mut(&config.partition) else {
+            return;
+        };
+
+        for slot in config.slots() {
+            buckets[slot].retain(|c| c.subscription_id != subscription_id);
         }
     }
 
-    /// Get all check configs that are scheduled for a given tick.
-    pub fn get_configs(&self, tick: Tick) -> Vec<Arc<CheckConfig>> {
-        self.buckets[tick.index]
+    /// Drop an entire partition of check configurations.
+    pub fn drop_partition(&mut self, partition: i32) {
+        let Some(buckets) = self.partitioned_buckets.remove(&partition) else {
+            return;
+        };
+
+        buckets
             .iter()
-            .filter_map(|id| self.configs.get(id).cloned())
+            .flat_map(|b| b.iter().cloned())
+            .for_each(|config| {
+                self.configs.remove(&config.subscription_id);
+            });
+    }
```

Comment:
This is one reason I'd like to not have the configs be global - ideally when we drop a partition we don't need to do anything other than just remove it from some global mapping.